scripts/core/exp-bindless-images.yml

```diff
@@ -578,7 +578,10 @@ class: $xBindlessImages
 name: ImageGetInfoExp
 ordinal: "0"
 params:
-    - type: $x_exp_image_mem_handle_t
+    - type: $x_context_handle_t
+      name: hContext
+      desc: "[in] handle of the context object"
```

Comment:
Why is the context parameter being added here? No implementations use the context, and the extra parameter doesn't seem to be reflected in all of the adapters code. Additionally, if this change is necessary in any way, it will require changes on the DPC++ side too.

Reply:
The UR loader wraps every `_handle_t` type to insert a pointer to the adapter that created the handle. This way the loader is able to dispatch UR function calls to the correct adapter. This breaks DPC++'s usage of these image handles because it reinterpret casts them straight to pointers etc. This PR makes them native handles to disable the handle wrapping in the loader. However that means that `ImageGetInfoExp` no longer has any 'real' handles so the loader cannot dispatch the function, which is why the context handle has been added. It could alternatively be some other handle we know is available like the platform or adapter.

I realise this is a slightly awkward and disruptive change but it's the easiest fix I can come up with to fix bindless images for https://github.com/intel/llvm/pull/14145 - happy to take other suggestions but this is probably easiest in the short term

I'll put a intel/llvm PR up shortly.

Reply:
I see.

Does the context handle have to be valid? In which case we need to change the SYCL API to receive a context handle, and any code using this API. Or can we just pass a `nullptr` to UR since the value is never used by any backend?

Reply:
The handle does need to be valid because `nullptr` doesn't have any reference to an adapters dispatch table containing function pointers to the implementation of any UR entry point.

Reply:
@przemektmalon I've just put up a PR for DPC++ - https://github.com/intel/llvm/pull/14516 - the SYCL API doesn't need changed because the implementation already has access to the context any time ImageGetInfo is used

Reply:
> the SYCL API doesn't need changed because the implementation already has access to the context any time ImageGetInfo is used

Ah, you're right. I left two minor comments on the DPC++ PR. Everything LGTM otherwise.